src/tuning/tuning.cpp

```diff
@@ -89,6 +91,106 @@ void print_separator(const size_t parameters_size) {
 
 // =================================================================================================
 
+struct ThreadInfo {
+  std::mutex mtx;
+  std::string print_info;
+  Kernel kernel{nullptr};
+  bool ready = false;
+  std::condition_variable cv;
+  std::vector<size_t> global;
+  std::vector<size_t> local;
+};
+
+template <typename... Args>
+inline void addPrintInfo(std::string& str, const char* format, Args&&... args) {
+  char temp[] = "m";
+  auto size = std::snprintf(temp, 1, format, std::forward<Args>(args)...);
+  auto original_size = str.size();
+  str.resize(original_size + size);
+  std::snprintf(&str[original_size], size + 1, format, std::forward<Args>(args)...);
+}
+
+template <typename T>
+void tuningThread(std::vector<ThreadInfo>& infos, const std::vector<clblast::Configuration>& configurations, size_t id,
+                  const TunerSettings& settings, const Arguments<T>& args, const Device& device, const Context& context,
+                  const size_t num_threads) {
+#if defined(_WIN32)
+  const std::string kPrintError = "";
+  const std::string kPrintSuccess = "";
+  const std::string kPrintMessage = "";
+  const std::string kPrintEnd = "";
+#else
+  const std::string kPrintError = "\x1b[31m";
+  const std::string kPrintSuccess = "\x1b[32m";
+  const std::string kPrintMessage = "\x1b[1m";
+  const std::string kPrintEnd = "\x1b[0m";
+#endif
+  for (size_t config_id = id; config_id < configurations.size(); config_id += num_threads) {
+    auto& info = infos[config_id];
+    info.mtx.lock();
+    try {
+      auto configuration = configurations[config_id];
+      addPrintInfo(info.print_info, "| %4zu | %5zu |", config_id + 1, configurations.size());
+
+      for (const auto& parameter : settings.parameters) {
+        addPrintInfo(info.print_info, "%5zu", configuration.at(parameter.first));
+      }
+      addPrintInfo(info.print_info, " |");
+
+      // Sets the thread configuration
```

Comment:
To avoid confusion with the C++ compilation thread:
```suggestion
      // Sets the OpenCL thread configuration
```